.github/workflows/interface-unit-tests.yml

```diff
@@ -26,14 +26,127 @@ on:
         required: false
         type: string
         default: --cov=pennylane --cov-report=term-missing --cov-report=xml --no-flaky-report -p no:warnings --tb=native
+      run_lightened_ci:
+        description: |
+          Indicate if a lightened version of the CI should be run instead of the entire suite.
+          
+          The lightened version of the CI includes the following changes:
+          - Only Python 3.9 is tested against, instead of 3.9, 3.10, 3.11
+        required: false
+        type: boolean
+        default: false
+      skip_ci_test_jobs:
+        description: |
+          Names of jobs (comma separated) that should be skipped on a lightened CI run.
+          The value of this variable is only used IF 'run_lightened_ci' is `true`.
+          For a full test-suite run, all jobs are triggered.
+        required: false
+        type: string
+        default: ''
 
 jobs:
+  setup-ci-load:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup Python Versions
+        id: python_versions
+
+        # All jobs will use the 'default' python versions listed in the dictionary below.
+        # Unless the job name exists as a key as well, in which case the python versions listed for the job itself will be used instead.
+        run: |
+          if [ "${{ inputs.run_lightened_ci }}" == "true" ]; 
+          then
+            cat >python_versions.json <<-EOF
+          {
+            "default": ["3.9"]
+          }
+          EOF
+          else
+            cat >python_versions.json <<-EOF
+          {
+            "default": ["3.9", "3.10", "3.11"],
+            "torch-tests": ["3.9", "3.11"],
+            "tf-tests": ["3.9", "3.10"],
+            "jax-tests": ["3.9", "3.11"],
+            "all-interfaces-tests": ["3.9"],
+            "external-libraries-tests": ["3.9"],
+            "qcut-tests": ["3.9"],
+            "qchem-tests": ["3.9"],
+            "gradients-tests": ["3.9"],
+            "data-tests": ["3.10"],
+            "device-tests": ["3.9"]
+          }
+          EOF
+          fi
+
+          jq . python_versions.json
+          echo "python_versions=$(jq -r tostring python_versions.json)" >> $GITHUB_OUTPUT
```

Comment:
is this the best way to save dicts between job stages in github CI? I mean.. it gets the job (sorry) done lol